external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaRecordTupleBuilder.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ *   or more contributor license agreements.  See the NOTICE file
+ *   distributed with this work for additional information
+ *   regarding copyright ownership.  The ASF licenses this file
+ *   to you under the Apache License, Version 2.0 (the
+ *   "License"); you may not use this file except in compliance
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package org.apache.storm.kafka.spout;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Values;
+
+import java.util.List;
+
+public class KafkaRecordTupleBuilder<K, V> implements KafkaSpoutTupleBuilder<K, V> {
+    @Override
+    public List<Object> buildTuple(final ConsumerRecord<K, V> consumerRecord, KafkaSpoutStreams kafkaSpoutStreams) {
```

Comment:
There are three entities being constructed 
1. Tuple - This is a list of objects being emitted by the spout e.g. ["electronics", "abhishek", "router"]
2. OutputFields - this indicates the position of fields by their name e.g. [CATEGORY_NAME, USER_NAME, ITEM_NAME]
The bolt uses the field name in OutputFields to access the value in tuple. e.g. one might access category using `String category = tuple.getStringByField(CATEGORY_NAME)`
Person who is constructing the tuple, knows which positions of different fields in tuple. That's why I suggested to have tuple building and output fields in single interface because these two are tied and need to be consistent with each other.

3. Schema - *a subset of the output fields* - I am not sure what it is. can you elaborate more on this?